### PR TITLE
feat: Battery service implementation + tests (all 15 opcodes)

### PR DESCRIPTION
**Pull request by dymk**
## Summary

Implement all 15 battery opcodes (BTP, BMC, BST, BIX, etc.) with stateful
Battery struct. Adds 20+ unit tests covering all opcodes, edge cases, and
the BTP/BMC zero-value write fix.

## Changes
- `platform/qemu-sp/src/battery.rs`: Battery struct with state tracking, all 15 opcode handlers, comprehensive test module

## Opcodes Implemented
BTP, BMC, BST, BIX, BPC, BPS, BCT, BCV, BAR, BAT_DIS, BAT_STA, BAT_PSR, BAT_SOC, BAT_CYC, BAT_FCC

## Key Design Decisions
- Uses `set_flag` byte mechanism for BTP/BMC to distinguish 'set to zero' from 'no update'
- All state stored in stack-based Battery struct (no_std compatible)
- Response serialization via fixed-size byte array packing

### CODE REVIEW

File: platform/qemu-sp/src/battery.rs [hunk 1]
```diff
@@ -57,25 +46,313 @@ impl From<&DirectMessagePayload> for BstRsp {
...
+impl From<&DirectMessagePayload> for ValueRsp {
+    fn from(payload: &DirectMessagePayload) -> Self {
+        ValueRsp {
+            value: payload.u32_at(0),
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub struct Battery {
```
Comment (philgweber) [inline platform/qemu-sp/src/battery.rs:173]: Generally I'm not sure we want to keep cached values of all fields on a device that has an actual battery, we would probably each time get updated state from the EC. For stub code it makes sense so we can keep the last values set. Maybe we can move this to a separate stub file? I'd love if we can reuse the same battery.rs to make the actual calls to battery and if we compile for stubs it returns the stub data?

Comment (dymk) [reply platform/qemu-sp/src/battery.rs:173]: @philgweber I'd rather keep the code here very simple, and replace the whole thing here with real code that talks to the EC over a comms interface once platforms repo has the qemu based EC code available to it

**Review by philgweber — DISMISSED** [review]

**Review by kurtjd — DISMISSED** [review]

**Review by philgweber — DISMISSED** [review]

**Review by kurtjd — DISMISSED** [review]

**Review by rogurr — DISMISSED** [review]
